tests/agent/test_prompt.py

```
@@ -0,0 +1,250 @@
+    else:
+        raise ValueError(f"Unknown provider: {config['provider']}")
+
+    return extract_cuda_code(response)
+
+
+def save_test_results(
+    output_file: Path,
+    model_name: str,
+    cuda_code: str,
+    compilation_success: bool,
+    error_msg: str = None,
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)